base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs

```diff
@@ -88,28 +89,24 @@ where
     ) -> Result<Vec<RecoveredOutput>, OutputManagerError> {
         let start = Instant::now();
         let outputs_length = outputs.len();
-        let mut rewound_outputs: Vec<(UnblindedOutput, BulletRangeProof)> = outputs
-            .into_iter()
-            .filter_map(|output| {
-                output
-                    .full_rewind_range_proof(
-                        &self.factories.range_proof,
-                        &self.rewind_data.rewind_key,
-                        &self.rewind_data.rewind_blinding_key,
-                    )
-                    .ok()
-                    .map(|v| (v, output))
-            })
-            .filter_map(|(rewind_result, output)| {
-                if output.script != script!(Nop) {
-                    return None;
-                }
-                let script_key = PrivateKey::random(&mut OsRng);
-                Some((
-                    UnblindedOutput::new(
+
+        let mut rewound_outputs: Vec<(UnblindedOutput, BulletRangeProof)> = Vec::new();
+        for output in outputs {
+            // TODO: Only outputs with scripts `== script!(Nop)` is recover-able - can this be improved?
+            if output.script != script!(Nop) {
```

Comment:
Is this really the case?

Reply:
The previous version of the code had this exact check - if a rewound was successful but the script was not `script!(Nop)` it dropped the output. Maybe this is latent defect? That is why I logged a `TODO`, but I will also register an issue.